### PR TITLE
Merge v5.1.1 into `next`

### PR DESCRIPTION
**Pull request by ZedThree**
There were quite a lot of conflicts in the SUNDIALS solvers, so this definitely needs double-checking

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,67 @@
...
+- CI: Run on PETSc developement branch [\#2765][https://github.com/boutproject/BOUT-dev/pull/2765] ([dschwoerer][https://github.com/dschwoerer])
+- Prefer dnf5 for fedora CI (master) [\#2764][https://github.com/boutproject/BOUT-dev/pull/2764] ([dschwoerer][https://github.com/dschwoerer])
+- Fix sphinx configuration [\#2762][https://github.com/boutproject/BOUT-dev/pull/2762] ([dschwoerer][https://github.com/dschwoerer])
+- Bump docker/build-push-action from f2a1d5e99d037542a71f64918e516c093c6f3fc4 to 0f847266c302569530c95bfa228489494c43b002 [\#2761][https://github.com/boutproject/BOUT-dev/pull/2761] ([dependabot[bot]][https://github.com/dependabot%5Bbot%5D])
+- Ensure also lower case is accepted [\#2759][https://github.com/boutproject/BOUT-dev/pull/2759] ([dschwoerer][https://github.com/dschwoerer])
+- Bump externalpackages/googletest from `6092810` to `2dd1c13` [\#2757][https://github.com/boutproject/BOUT-dev/pull/2757] ([dependabot[bot]][https://github.com/dependabot%5Bbot%5D])
+- Add fallback if the version has been bumped bot not tagged [\#2754][https://github.com/boutproject/BOUT-dev/pull/2754] ([dschwoerer][https://github.com/dschwoerer])
```
Comment (dschwoerer) [inline CHANGELOG.md:63]:
```suggestion
- Add fallback if the version has been bumped but not tagged [\#2754][https://github.com/boutproject/BOUT-dev/pull/2754] ([dschwoerer][https://github.com/dschwoerer])
```

File: CITATION.cff [hunk 2]
```diff
@@ -188,11 +188,20 @@ authors:
...
+    given-names: Kefu
+  - family-names: Roberts
+    given-names: Steven
+  - family-names: Georgakoudis
+    given-names: Giorgis
+  - family-names: Kryjak
+    given-names: Mike
+
+version: 5.1.1
 date-released: 2023-04-10
```
Comment (dschwoerer) [inline CITATION.cff:201]: that should be updated?

File: locale/de/libbout.po [hunk 3]
```diff
@@ -273,28 +285,33 @@ msgstr "\tKontravariantevektoren werden gelesen "
...
+#: ../src/bout++.cxx:574
+#, c++-format
 msgid "\tSUNDIALS support {}\n"
 msgstr ""
 
-#: ../src/bout++.cxx:571
+#: ../src/bout++.cxx:572
+#, c++-format
 msgid "\tScore-P support {}\n"
 msgstr ""
```
Comment (dschwoerer) [inline locale/de/libbout.po:306]: We probably should remove empty translations?

File: src/solver/impls/cvode/cvode.cxx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/**************************************************************************
+n/**************************************************************************
```
Comment (dschwoerer) [inline src/solver/impls/cvode/cvode.cxx:1]:
```suggestion
/**************************************************************************
```

CI should have caught this one!

**Review by dschwoerer — CHANGES_REQUESTED** [review]
I am worried that the CI did not catch the `n`.
Can we rerun the CI?

**Comment by ZedThree** [comment]
@Steven-Roberts Would you be able to double-check the SUNDIALS parts have been resolved correctly please?

**Comment by Steven-Roberts** [comment]
I can take a look early next week

**Comment by Steven-Roberts** [comment]
@ZedThree The SUNDIALS-related changes look fine. I confirmed that it compiles and an example can run using SUNDIALS version 4, 5, 6, and 7.

**Comment by bendudson** [comment]
Thanks @Steven-Roberts for checking the SUNDIALS changes!

**Pull request merged**